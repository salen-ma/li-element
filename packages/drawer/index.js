import LiDrawer from './src/drawer.vue'

LiDrawer.install = Vue => {
  Vue.component(LiDrawer.name, LiDrawer)
}

export default LiDrawer