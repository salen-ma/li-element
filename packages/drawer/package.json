{
  "name": "li-drawer",
  "version": "0.0.1",
  "main": "index.js",
  "license": "MIT"
}
