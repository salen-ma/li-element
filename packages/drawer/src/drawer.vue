<template>
  <div
    class="li-drawer"
    :class="[
      `li-drawer--${placement}`,
      {
        'li-drawer--open': visible
      }
    ]"
  >
    <div class="li-drawer-mask"></div>
    <div 
      class="li-drawer-content" 
      :style="{
        width: width + 'px'
      }"
    >
    </div>
    <slot></slot>
  </div>
</template>

<script>
export default {
  name: 'LiDrawer',
  props: {
    visible: {
      type: Boolean,
      default: false
    },
    placement: {
      type: String,
      default: 'right'
    },
    width: {
      type: Number,
      default: 500
    }
  },
}
</script>

<style lang='scss'>
.li-drawer {
  position: fixed;
  z-index: 1000; 
  &--top {
    top: 0;
    .li-drawer-content {
      top: 0;
    }
  }
  &--right {
    right: 0;
    .li-drawer-content {
      right: 0;
    }
  }
  &--bottom {
    bottom: 0;
    .li-drawer-content {
      bottom: 0;
    }
  }
  &--left {
    left: 0;
    .li-drawer-content {
      left: 0;
    }
  }
  &--left, &--right{
    top: 0;
    height: 100%;
  }
  &--top, &--bottom{
    left: 0;
    width: 100%;
  }
  &--open {
    width: 100%;
    .li-drawer-mask {
      height: 100%;
      opacity: 1;
      -webkit-transition: none;
      transition: none;
      -webkit-animation: antdDrawerFadeIn .3s cubic-bezier(.7,.3,.1,1);
      animation: antdDrawerFadeIn .3s cubic-bezier(.7,.3,.1,1);
      pointer-events: auto;
    }
  }
}
.li-drawer-mask {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 0;
  background-color: rgba(0,0,0,.45);
  opacity: 0;
  filter: alpha(opacity=45);
  -webkit-transition: opacity .3s linear,height 0s ease .3s;
  transition: opacity .3s linear,height 0s ease .3s;
  pointer-events: none;
}
.li-drawer-content {
  position: absolute;
  background: #ffffff;
  height: 100%;
}
</style>
