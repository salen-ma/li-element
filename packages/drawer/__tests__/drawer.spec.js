import drawer from '../src/drawer.vue'
import { shallowMount } from '@vue/test-utils'

describe('li-drawer', () => {
  test('drawer title', () => {
    const wrapper = shallowMount(drawer, {
      propsData: {
        title: "basic drawer"
      }
    })
    expect(wrapper.html()).toContain('basic drawer')
  })

  test('drawer snapshot', () => {
    const wrapper = shallowMount(drawer, {
      propsData: {
        title: "basic drawer",
        visible: true
      }
    })
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})