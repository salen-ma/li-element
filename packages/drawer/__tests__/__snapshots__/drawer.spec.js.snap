// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`li-drawer drawer snapshot 1`] = `
<transition-stub
  name="el-drawer-fade"
>
  <div
    class="li-drawer li-drawer--right li-drawer--open"
  >
    <div
      class="li-drawer-mask"
    />
     
    <div
      class="li-drawer-content"
      style="width: 300px; height: 100%;"
    >
      <div
        class="li-drawer-header"
      >
        
        basic drawer
      
      </div>
        
      <div
        class="li-drawer-body"
      />
       
      <!---->
    </div>
  </div>
</transition-stub>
`;
